substratevm/src/com.oracle.svm.jfr/src/com/oracle/svm/jfr/SubstrateJVM.java

```diff
@@ -80,7 +80,8 @@
         typeRepo = new JfrTypeRepository();
         methodRepo = new JfrMethodRepository(typeRepo, symbolRepo);
         stackTraceRepo = new JfrStackTraceRepository(methodRepo);
-        repositories = new JfrRepository[]{stringRepo, typeRepo, symbolRepo, methodRepo, stackTraceRepo};
+        JfrFrameTypeSerializer frameTypeSerializer = new JfrFrameTypeSerializer();
+        repositories = new JfrRepository[]{frameTypeSerializer, stringRepo, typeRepo, symbolRepo, methodRepo, stackTraceRepo};
```

Comment:
Does the ordering here affect the order at which the repositories are written? I think we want the `methodRepo` before the `symbolRepo` as it could add symbols during the process right? I don't know what's in the `stackTraceRepo` but also to consider I suppose.

Can a comment be added about the ordering? On another note, I'll need to revisit the repositories after my chunk rotation change as it moves them all to write after the epoch changes.

Reply:
> Does the ordering here affect the order at which the repositories are written?

Yes, it does.

> I think we want the `methodRepo` before the `symbolRepo` as it could add symbols during the process right? I don't know what's in the `stackTraceRepo` but also to consider I suppose.

Right now, methodRepo and stackTraceRepo are unused and don't write anything. I am not sure if any of them would use the symbolRepo. I've ordered the repos like I did here because that is how OpenJDK writes it, the symbols always seem to go together with classes/packages/modules/classLoaders. I'll investigate how methods and stacktraces are written when I get to it. They (can) probably emit another separate block of symbols or so, not sure.

My intention with this patch was to not change any functional aspects.

> Can a comment be added about the ordering?

Ok, will do.

> On another note, I'll need to revisit the repositories after my chunk rotation change as it moves them all to write after the epoch changes.

Yeah, they should be written after epoch changed.

Reply:
> > Does the ordering here affect the order at which the repositories are written?
> 
> Yes, it does.
> 
> > I think we want the `methodRepo` before the `symbolRepo` as it could add symbols during the process right? I don't know what's in the `stackTraceRepo` but also to consider I suppose.
> 
> Right now, methodRepo and stackTraceRepo are unused and don't write anything. I am not sure if any of them would use the symbolRepo. I've ordered the repos like I did here because that is how OpenJDK writes it, the symbols always seem to go together with classes/packages/modules/classLoaders. I'll investigate how methods and stacktraces are written when I get to it. They (can) probably emit another separate block of symbols or so, not sure.

The serialize method for the type set in jdk and jdk11 both call `write_methods` before `write_symbols`:
https://github.com/openjdk/jdk/blob/master/src/hotspot/share/jfr/recorder/checkpoint/types/jfrTypeSet.cpp#L1110
https://github.com/openjdk/jdk11u-dev/blob/master/src/hotspot/share/jfr/recorder/checkpoint/types/jfrTypeSet.cpp#L943

so I think at least the methods can be moved before symbols.

> 
> My intention with this patch was to not change any functional aspects.
> 
> > Can a comment be added about the ordering?
> 
> Ok, will do.
> 
> > On another note, I'll need to revisit the repositories after my chunk rotation change as it moves them all to write after the epoch changes.
> 
> Yeah, they should be written after epoch changed.